### ISSUE TITLE
Windows Launch Error

### ISSUE THREAD

**Issue by not-matt** [ahodges9/LedFx#2]
Install completed successfully (?), but upon running program gives error "failed to create process." with no further info.

**Comment by ahodges9** [comment]
Hmm, where is your python installed? Seems like this is common if there is a space in that path. It looks like this might be a setuptools bug but ill see if I can find a workaround. It should just be an issue with the script generation to make "ledfx" work. Running manually should work

Can you try to launch using:
  python -m ledfxcontroller --open-ui

**Comment by not-matt** [comment]
Yeah I've found a few places mentioning how a space in the install path can mess things up. Mine has a space. Upgraded pip and setuptools then reinstalled ledfx with no luck.

Running manually worked, but I then had to manually install the package "voluptuous".

Ran again, with a new error, but now the install is working at least!

**Comment by not-matt** [comment]
Full traceback for new error:

Loading configuration file from C:\Users\Matthew Bowley\AppData\Roaming\.ledfx
Traceback (most recent call last):
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\__main__.py", line 92, in <module>
    sys.exit(main())
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\__main__.py", line 88, in main
    ledfx = LedFxController(config_dir = args.config)
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\core.py", line 22, in __init__
    self.config = load_config(config_dir)
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\config.py", line 93, in load_config
    return CORE_CONFIG_SCHEMA(yaml.load(file))
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\site-packages\voluptuous\schema_builder.py", line 267, in __call__
    return self._compiled([], data)
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\site-packages\voluptuous\schema_builder.py", line 587, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\site-packages\voluptuous\schema_builder.py", line 425, in validate_mapping
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: expected dict for dictionary value @ data['devices']

**Comment by not-matt** [comment]
Just spotted this, popped up on first run.

"Failed to find configuration file. Creating default configuration file in None"

**Comment by ahodges9** [comment]
Okay, Ill dig a little more into in later. I figured we are going to hit a few things like this while we get the new environments. I missed voluptuous from the requirements so ill get that fixed.

For the missing config get_default_config_directory() should pull the correct one, but might have missed some environments. What does this produce:

import os
os.name

**Comment by not-matt** [comment]
Win, gives "nt"

**Comment by ahodges9** [comment]
Can you send me the output running with the -vv flag?

**Comment by not-matt** [comment]
...
# trying C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\DLLs\colorsys.pyc
# trying C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\colorsys.cp36-win32.pyd
# trying C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\colorsys.pyd
# trying C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\colorsys.py
# C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\__pycache__\colorsys.cpython-36.pyc matches C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\colorsys.py
# code object from 'C:\\Users\\Matthew Bowley\\AppData\\Local\\Programs\\Python\\Python36-32\\lib\\__pycache__\\colorsys.cpython-36.pyc'
import 'colorsys' # <_frozen_importlib_external.SourceFileLoader object at 0x08A98C70>
import 'ledfxcontroller.effects' # <_frozen_importlib_external.SourceFileLoader object at 0x08A7B3B0>
# trying C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\config.cp36-win32.pyd
# trying C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\config.pyd
# trying C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\config.py
# C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\__pycache__\config.cpython-36.pyc matches C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\config.py
# code object from 'C:\\Users\\Matthew Bowley\\Documents\\Python\\LedFx\\ledfxcontroller\\__pycache__\\config.cpython-36.pyc'
import 'ledfxcontroller.config' # <_frozen_importlib_external.SourceFileLoader object at 0x097F6170>
import 'ledfxcontroller.core' # <_frozen_importlib_external.SourceFileLoader object at 0x04229A10>
Loading configuration file from C:\Users\Matthew Bowley\AppData\Roaming\.ledfx
Traceback (most recent call last):
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\__main__.py", line 92, in <module>
    sys.exit(main())
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\__main__.py", line 88, in main
    ledfx = LedFxController(config_dir = args.config)
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\core.py", line 22, in __init__
    self.config = load_config(config_dir)
  File "C:\Users\Matthew Bowley\Documents\Python\LedFx\ledfxcontroller\config.py", line 93, in load_config
    return CORE_CONFIG_SCHEMA(yaml.load(file))
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\site-packages\voluptuous\schema_builder.py", line 267, in __call__
    return self._compiled([], data)
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\site-packages\voluptuous\schema_builder.py", line 587, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "C:\Users\Matthew Bowley\AppData\Local\Programs\Python\Python36-32\lib\site-packages\voluptuous\schema_builder.py", line 425, in validate_mapping
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: expected dict for dictionary value @ data['devices']
# clear builtins._
# clear sys.path
# clear sys.argv
# clear sys.ps1
# clear sys.ps2
...

**Comment by ahodges9** [comment]
Ahh I see what's going on. I changes the config format a bit and didn't update the autogen code. I'll get a fix soon.

**Comment by not-matt** [comment]
Good stuff! Looking forward to trying this out.

**Comment by not-matt** [comment]
Fixed the config file manually.

Module win32api was also missing and was a pain in the ass to install through pip. Got it from [here](https://github.com/mhammond/pywin32/releases) instead and it works fine.

**Comment by ahodges9** [comment]
Keeping this active until the various issues you pointed out are fixed. There shouldn't be manual steps and you should be able to easily install and run regardless of setup.

Also there isn't much to the apps UI yet as I'm working on the core infrastructure. Essentially just proof of concept currently :)
